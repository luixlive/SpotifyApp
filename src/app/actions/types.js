// Device reducer
export const DEVICE_TYPE_CHANGED = 'DEVICE_TYPE_CHANGED';

// User
export const LOAD_USER = 'LOAD_USER';
export const LOAD_USER_FAILED = 'LOAD_USER_FAILED';
export const LOAD_USER_SUCCEEDED = 'LOAD_USER_SUCCEEDED';
export const LOGOUT_USER = 'LOGOUT_USER';
export const LOGOUT_USER_FAILED = 'LOGOUT_USER_FAILED';
export const LOGOUT_USER_SUCCEEDED = 'LOGOUT_USER_SUCCEEDED';

// UserStats
export const LOAD_USER_STATS = 'LOAD_USER_STATS';
export const LOAD_USER_STATS_FAILED = 'LOAD_USER_STATS_FAILED';
export const LOAD_USER_STATS_SUCCEEDED = 'LOAD_USER_STATS_SUCCEEDED';
