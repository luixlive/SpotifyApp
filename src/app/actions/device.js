import { DEVICE_TYPE_CHANGED } from './types';

export default deviceMobile => (
  { type: DEVICE_TYPE_CHANGED, payload: { deviceMobile } }
);
