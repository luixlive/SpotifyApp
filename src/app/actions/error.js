import { CLEAR_ERROR } from './types';

export default () => (
  { type: CLEAR_ERROR, payload: {} }
);
