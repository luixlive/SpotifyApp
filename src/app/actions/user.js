import { LOAD_USER, LOGOUT_USER } from './types';

export const loadUser = () => (
  { type: LOAD_USER, payload: { } }
);

export const logoutUser = () => (
  { type: LOGOUT_USER, payload: { } }
);
