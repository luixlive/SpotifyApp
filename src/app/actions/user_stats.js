import { LOAD_USER_STATS } from './types';

export default () => (
  { type: LOAD_USER_STATS, payload: { } }
);
