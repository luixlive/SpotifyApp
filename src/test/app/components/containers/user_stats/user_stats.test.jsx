import _ from 'lodash';
import configureStore from 'redux-mock-store';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';

import initialState from './../../../../test_utils/initial_state';
import {
  Artists,
  Profile,
  Tracks,
} from './../../../../../app/components/containers/user_stats';
import * as types from './../../../../../app/actions/types';
import UserStats, {
  PureUserStats,
} from './../../../../../app/components/containers/user_stats/user_stats';
import {
  userStats as componentProps,
} from './../../../../test_utils/components_props';

describe('App Components - UserStats', () => {
  describe('Snapshots', () => {
    const mockStore = configureStore();
    let props;
    let store;
    beforeAll(() => {
      props = _.cloneDeep(componentProps);
      store = mockStore(initialState);
    });

    it('renders stats not loaded', () => {
      const rendered = renderer.create(<PureUserStats {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders stats loaded', () => {
      props.statsLoaded = true;
      const rendered = renderer.create((
        <Provider store={store}>
          <PureUserStats {...props} />
        </Provider>
      ));
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    let props;
    beforeAll(() => {
      props = _.cloneDeep(componentProps);
    });

    it('calls loadUserStats when componentDidMount', () => {
      const loadUserStats = jest.fn();
      mount(<PureUserStats {...props} loadUserStats={loadUserStats} />);
      expect(loadUserStats).toHaveBeenCalledTimes(1);
    });

    it('updates state.activeMenuItem when clicking menu items', () => {
      props.statsLoaded = true;
      const wrapper = shallow(<PureUserStats {...props} />);
      expect(wrapper.state().activeMenuItem).toBe(PureUserStats.PROFILE);
      wrapper.find('MenuItem').at(1).simulate('click');
      expect(wrapper.state().activeMenuItem).toBe(PureUserStats.TRACKS);
      wrapper.find('MenuItem').at(2).simulate('click');
      expect(wrapper.state().activeMenuItem).toBe(PureUserStats.ARTISTS);
    });

    it('renders Profile as default', () => {
      props.statsLoaded = true;
      const wrapper = shallow(<PureUserStats {...props} />);
      expect(wrapper.find(Profile).length).toBe(1);
    });

    it('renders Profile if option in menu is clicked', () => {
      props.statsLoaded = true;
      const wrapper = shallow(<PureUserStats {...props} />);
      wrapper.find('MenuItem').at(1).simulate('click');
      expect(wrapper.find(Profile).length).toBe(0);
      wrapper.find('MenuItem').at(0).simulate('click');
      expect(wrapper.find(Profile).length).toBe(1);
    });

    it('renders Tracks if option in menu is clicked', () => {
      props.statsLoaded = true;
      const wrapper = shallow(<PureUserStats {...props} />);
      expect(wrapper.find(Tracks).length).toBe(0);
      wrapper.find('MenuItem').at(1).simulate('click');
      expect(wrapper.find(Tracks).length).toBe(1);
    });

    it('renders Artists if option in menu is clicked', () => {
      props.statsLoaded = true;
      const wrapper = shallow(<PureUserStats {...props} />);
      expect(wrapper.find(Artists).length).toBe(0);
      wrapper.find('MenuItem').at(2).simulate('click');
      expect(wrapper.find(Artists).length).toBe(1);
    });

    it('does not render anything in a weird behavior', () => {
      props.statsLoaded = true;
      const wrapper = shallow(<PureUserStats {...props} />);
      wrapper.setState({ activeMenuItem: 'unknown' });
      expect(wrapper.find(Profile).length).toBe(0);
      expect(wrapper.find(Tracks).length).toBe(0);
      expect(wrapper.find(Artists).length).toBe(0);
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeEach(() => {
      store = mockStore(initialState);
      wrapper = mount((
        <Provider store={store}>
          <UserStats />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(PureUserStats).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(Object.keys(wrapper.find(PureUserStats).props()))
        .toEqual(Object.keys(componentProps));
    });

    it('dispatches loadUserStats', () => {
      const expectedAction = {
        type: types.LOAD_USER_STATS,
        payload: {},
      };
      expect(store.getActions()[0]).toEqual(expectedAction);
    });
  });
});
