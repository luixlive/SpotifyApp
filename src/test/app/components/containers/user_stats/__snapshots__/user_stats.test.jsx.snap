// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - UserStats Snapshots renders stats loaded 1`] = `
<div>
  <div
    className="ui pointing secondary three item menu"
  >
    <a
      className="active item"
      onClick={[Function]}
    >
      
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Tracks
    </a>
    <a
      className="item"
      onClick={[Function]}
    >
      Artists
    </a>
  </div>
  <div>
    <img
      alt="hola"
      className="ui medium circular centered image"
      onError={[Function]}
      src="url"
    />
    <div
      className="ui centered card"
      href={undefined}
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="header"
        >
          
        </div>
        <div
          className="meta"
        >
          <span>
            
          </span>
        </div>
        <div
          className="description"
        >
          <a
            href={undefined}
            target="_blank"
          >
            See in Spotify
          </a>
        </div>
      </div>
      <div
        className="extra content"
      >
         followers
      </div>
    </div>
  </div>
</div>
`;

exports[`App Components - UserStats Snapshots renders stats not loaded 1`] = `
<div
  className="ui basic segment body"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui big inverted text loader"
        >
          <h3
            className="ui header"
          >
            Loading...
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
